pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -71,66 +80,106 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctr
-			return e
+	// Build communal storage params if there is not one
+	if q.ConfigurationParams == nil {
+		res, err = q.ConstructConfigParms(ctx)
+		if verrors.IsReconcileAborted(res, err) {
+			return res, err
 		}
-		q.Vdb = vdb
-		// build communal storage params if there is not one
-		if q.ConfigurationParams == nil {
-			res, e = q.ConstructConfigParms(ctx)
-			if verrors.IsReconcileAborted(res, e) {
-				return e
-			}
-		}
-		// extract out the communal and config information to pass down to the vclusterops API.
-		opts = append(opts,
-			restorepointsquery.WithCommunalPath(vdb.GetCommunalPath()),
-			restorepointsquery.WithConfigurationParams(q.ConfigurationParams.GetMap()),
-		)
-
-		return nil
-	})
+	}
 
 	return res, err
 }
 
-// runListRestorePoints will update the status condition before and after calling
-// list restore points api
-// Temporarily, runListRestorePoints will not call the ListRestorePoints API
-// since the dispatcher is not set up yet
-func (q *QueryReconciler) runListRestorePoints(ctx context.Context, _ *ctrl.Request) error {
+// runShowRestorePoints call the vclusterOps API to get the restore points
+func (q *QueryReconciler) runShowRestorePoints(ctx context.Context, dispatcher vadmin.Dispatcher,
+	opts []restorepointsquery.Option) (err error) {
 	// set Querying status condition and state prior to calling vclusterops API
-	err := vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+	err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
 		v1.MakeCondition(vapi.Querying, metav1.ConditionTrue, "Started"), stateQuerying)
 	if err != nil {
 		return err
 	}
 
-	// API should be called to proceed here
-	// If we receive a failure result from the API, a state message and condition need to be updated
-
-	// clear Querying status condition
-	err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
-		v1.MakeCondition(vapi.Querying, metav1.ConditionFalse, "Completed"), stateQuerying)
-	if err != nil {
-		return err
+	// call showRestorePoints vcluster API
+	q.VRec.Eventf(q.Vrpq, corev1.EventTypeNormal, events.ShowRestorePointsStarted,
+		"Starting show restore points")
+	start := time.Now()
+	if res, errRun := dispatcher.ShowRestorePoints(ctx, opts...); verrors.IsReconcileAborted(res, errRun) {
+		if errRun != nil {
+			err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+				v1.MakeCondition(vapi.Querying, metav1.ConditionFalse, "Failed"), stateFailedQuery)
+			if err != nil {
+				errRun = errors.Join(errRun, err)
+			}
+			return errRun
+		}
 	}
+	q.VRec.Eventf(q.Vrpq, corev1.EventTypeNormal, events.ShowRestorePointsSucceeded,
+		"Successfully queried restore points in %s", time.Since(start).Truncate(time.Second))
 
 	// set the QueryComplete if the vclusterops API succeeded
 	return vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
```

Comment:
We don't clear the Querying status condition if successful.

Reply:
oh I think you had a previous comments `Could we merge the two UpdateConditionAndState calls into one? We may need to allow multiple status conditions to be set?`

Reply:
You added it back, but now we have 2 status condition updates back to back. I was hoping to include this in a single status update. Maybe defer this until your next PR when you have to update the status with the results that come back. All 3 things, can be done in a single status update call.

Reply:
Sure, will keep in mind and fix it in part 2 for this PR